drivers/hmis/spec/requests/hmis/client_active_enrollment_spec.rb

```diff
@@ -0,0 +1,143 @@
+  let!(:c2_e2) { create :hmis_hud_enrollment, data_source: ds1, client: c2, project: p1, entry_date: 2.months.ago, exit_date: 1.month.ago }
+
+  describe 'when the user has full access' do
+    let!(:access_control) { create_access_control(hmis_user, ds1) }
+
+    it 'resolves activeEnrollment on multiple clients correctly' do
+      response, result = post_graphql(dob: shared_client_dob, project_id: p1.id, date: 2.days.ago) { query }
+      expect(response.status).to eq(200), result.inspect
+      clients = result.dig('data', 'clientSearch', 'nodes').map(&:deep_symbolize_keys)
+      expect(clients).to contain_exactly(
+        # enrollment for c1 is included
+        a_hash_including(id: c1.id.to_s, activeEnrollment: a_hash_including(id: c1_e1.id.to_s)),
+        # enrollment for c2 is not included
+        a_hash_including(id: c2.id.to_s, activeEnrollment: nil),
+      )
+    end
+
+    it 'chooses one activeEnrollment if client has multiple' do
+      # duplicate enrollment
+      dup_c1_e1 = create(:hmis_hud_enrollment, data_source: ds1, client: c1, project: p1, entry_date: 2.weeks.ago)
+
+      response, result = post_graphql(dob: shared_client_dob, project_id: p1.id, date: 2.days.ago) { query }
+
+      expect(response.status).to eq(200), result.inspect
+      clients = result.dig('data', 'clientSearch', 'nodes').map(&:deep_symbolize_keys)
+      expect(clients).to contain_exactly(
+        a_hash_including(id: c1.id.to_s, activeEnrollment: a_hash_including(id: dup_c1_e1.id.to_s)),
+        a_hash_including(id: c2.id.to_s, activeEnrollment: nil),
+      )
+    end
+
+    it 'includes WIP enrollment as activeEnrollment' do
+      c1_e1.save_in_progress!
+      expect(c1_e1.ProjectID).to be_nil
+
+      response, result = post_graphql(dob: shared_client_dob, project_id: p1.id, date: 2.days.ago) { query }
+
+      expect(response.status).to eq(200), result.inspect
+      clients = result.dig('data', 'clientSearch', 'nodes').map(&:deep_symbolize_keys)
+      expect(clients).to contain_exactly(
+        a_hash_including(id: c1.id.to_s, activeEnrollment: a_hash_including(id: c1_e1.id.to_s)),
+        a_hash_including(id: c2.id.to_s, activeEnrollment: nil),
+      )
+    end
+
+    it 'minimizes n+1 queries' do
+      shared_client_dob = 50.years.ago
+      20.times do
+        c = create(:hmis_hud_client, data_source: ds1, dob: shared_client_dob)
+        create(:hmis_hud_enrollment, data_source: ds1, client: c, project: p1, entry_date: 1.week.ago)
+      end
+
+      expect do
+        _, result = post_graphql(dob: shared_client_dob, project_id: p1.id, date: 2.days.ago) { query }
+        clients = result.dig('data', 'clientSearch', 'nodes').map(&:deep_symbolize_keys)
+        expect(clients.size).to eq(20)
+        # make sure each of them resolved an activeEnrollment
+        expect(clients.map { |c| c[:activeEnrollment] }.compact.size).to eq(20)
+      end.to make_database_queries(count: 10..35)
```

Comment:
cool! ✨

Reply:
right! it's also helpful testing tool imo to reduce the query count to something that you know will fail, because then the failure error will print a list of all the db queries that were made.